internal/controller/variantautoscaling_controller.go

```diff
@@ -508,27 +508,95 @@ func (r *VariantAutoscalingReconciler) readAcceleratorConfig(ctx context.Context
+		return config, nil
 	}
 
-	// Then, try to get from ConfigMap with retry logic
+	// Try ConfigMap second
+	if config, err := r.getPrometheusConfigFromConfigMap(ctx); err != nil {
+		return nil, fmt.Errorf("failed to get config from ConfigMap: %w", err)
+	} else if config != nil {
+		logger.Log.Info("Using Prometheus configuration from ConfigMap")
+		return config, nil
+	}
+
+	// No configuration found
+	logger.Log.Warn("No Prometheus configuration found. Please set PROMETHEUS_BASE_URL environment variable or configure via ConfigMap")
+	return nil, fmt.Errorf("no Prometheus configuration found. Please set PROMETHEUS_BASE_URL environment variable or configure via ConfigMap")
+}
+
+func (r *VariantAutoscalingReconciler) getPrometheusConfigFromEnv() (*interfaces.PrometheusConfig, error) {
+	promAddr := os.Getenv("PROMETHEUS_BASE_URL")
+	if promAddr == "" {
+		return nil, nil // No config found, but not an error
+	}
+
+	logger.Log.Info("Using Prometheus address from environment variable -> ", "address: ", promAddr)
+	return utils.ParsePrometheusConfigFromEnv(), nil
+}
+
+func (r *VariantAutoscalingReconciler) getPrometheusConfigFromConfigMap(ctx context.Context) (*interfaces.PrometheusConfig, error) {
 	cm := corev1.ConfigMap{}
 	err := utils.GetConfigMapWithBackoff(ctx, r.Client, configMapName, configMapNamespace, &cm)
 	if err != nil {
-		logger.Log.Warn("Failed to get ConfigMap for Prometheus config after retries, using default - ", "error: ", err)
-	} else if promAddr, exists := cm.Data["PROMETHEUS_BASE_URL"]; exists && promAddr != "" {
-		logger.Log.Info("Using Prometheus address from ConfigMap -> ", "address: ", promAddr)
-		return promAddr, nil
+		return nil, fmt.Errorf("failed to get ConfigMap for Prometheus config: %w", err)
+	}
+
+	promAddr, exists := cm.Data["PROMETHEUS_BASE_URL"]
+	if !exists || promAddr == "" {
+		return nil, nil // No config found, but not an error
+	}
+
+	logger.Log.Info("Using Prometheus address from ConfigMap -> ", "address: ", promAddr)
+
+	// Create config from ConfigMap data
+	config := &interfaces.PrometheusConfig{
+		BaseURL: promAddr,
+	}
+
+	// Parse TLS configuration from ConfigMap (TLS is always enabled for HTTPS-only support)
+	config.InsecureSkipVerify = func() bool {
+		if v, exists := cm.Data["PROMETHEUS_TLS_INSECURE_SKIP_VERIFY"]; exists {
+			return v == "true"
+		}
+		return false
+	}()
+	config.CACertPath = func() string {
+		if v, exists := cm.Data["PROMETHEUS_CA_CERT_PATH"]; exists {
+			return v
+		}
+		return ""
+	}()
+	config.ClientCertPath = func() string {
+		if v, exists := cm.Data["PROMETHEUS_CLIENT_CERT_PATH"]; exists {
+			return v
+		}
+		return ""
+	}()
+	config.ClientKeyPath = func() string {
+		if v, exists := cm.Data["PROMETHEUS_CLIENT_KEY_PATH"]; exists {
+			return v
+		}
+		return ""
+	}()
+	config.ServerName = func() string {
+		if v, exists := cm.Data["PROMETHEUS_SERVER_NAME"]; exists {
+			return v
+		}
+		return ""
+	}()
+
```

Comment:
Remove repeated code. Can we do something like this:

```
get := func(key, def string) string {
	if v, ok := cm.Data[key]; ok {
		return v
	}
	return def
}

config.InsecureSkipVerify = get("PROMETHEUS_TLS_INSECURE_SKIP_VERIFY", "") == "true"
config.CACertPath        = get("PROMETHEUS_CA_CERT_PATH", "")
config.ClientCertPath    = get("PROMETHEUS_CLIENT_CERT_PATH", "")
config.ClientKeyPath     = get("PROMETHEUS_CLIENT_KEY_PATH", "")
config.ServerName        = get("PROMETHEUS_SERVER_NAME", "")
```

Reply:
Thanks